bin/test/i18n/resources/dashboard/test_course_offerings.rb

```diff
@@ -3,17 +3,20 @@
 
 class I18n::Resources::Dashboard::CourseOfferingsTest < Minitest::Test
   def test_sync_in
-    CourseOffering.stubs(:all).returns(
-      [
-        CourseOffering.new(key: 'test_course_offering_key_2', display_name: 'test_course_offering_name_2'),
-        CourseOffering.new(key: 'test_course_offering_key_1', display_name: 'test_course_offering_name_1')
-      ]
-    )
+    CourseOffering.transaction do
+      CourseOffering.delete_all
 
-    expected_sync_in_result_file_path = CDO.dir('i18n/locales/source/dashboard/course_offerings.json')
-    expected_course_offerings_json = "{\n  \"test_course_offering_key_2\": \"test_course_offering_name_2\",\n  \"test_course_offering_key_1\": \"test_course_offering_name_1\"\n}"
-    File.expects(:write).with(expected_sync_in_result_file_path, expected_course_offerings_json).once
+      FactoryBot.create(:course_offering, key: 'course-offering-key-2', display_name: 'course-offering-name-2')
+      FactoryBot.create(:course_offering, key: 'course-offering-key-1', display_name: 'course-offering-name-1')
 
-    I18n::Resources::Dashboard::CourseOfferings.sync_in
+      File.expects(:write).with(
+        CDO.dir('i18n/locales/source/dashboard/course_offerings.json'),
+        %Q[{\n  "course-offering-key-2": "course-offering-name-2",\n  "course-offering-key-1": "course-offering-name-1"\n}]
```

Comment:
not super familiar with this sort of testing, I'd love to hear more details about how this test works.

Reply:
We `expect` the method `File.write` is going to be called `once` with 2 specified arguments during the execution of `I18n::Resources::Dashboard::CourseOfferings.sync_in`, otherwise the test is failed.
 https://github.com/freerange/mocha

Reply:
We `expects` the method `File.write` is going to be called `with` 2 specified arguments `once` during the execution of `I18n::Resources::Dashboard::CourseOfferings.sync_in`, otherwise the test is failed.
https://github.com/freerange/mocha